developer/src/kmc-model/src/build-trie.ts

```diff
@@ -491,18 +491,21 @@ namespace Trie {
  *
  * @param filename filename of the file to detect encoding
  */
-function detectEncoding(filename: string): 'utf8' | 'utf16le' {
-  let buffer = readFileSync(filename);
+function detectEncoding(buffer: Uint8Array): 'utf-8' | 'utf-16le' {
```

Comment:
TextDecoder only accepts `utf-8` and `utf-16le` unlike `fs.readFile`

Reply:
yup https://nodejs.org/docs/latest-v18.x/api/buffer.html#buffers-and-character-encodings if you want a reference